packages/components/src/button/stories/index.js

```diff
@@ -1,8 +1,3 @@
-/**
- * External dependencies
- */
-import React from 'react';
```

Comment:
If this works then we can remove the ignore from `eslint`.

Reply:
I think we still need this override in `eslint` because Storybook runs in development mode and will have development dependencies defined in the `package.json` in the root folder. 

`import/no-extraneous-dependencies` with the current setup ensure that production code has all dependencies defined in the packages that use them.

Reply:
Yeah, but none of those dev dependencies get explicitly imported in story files so we would be good.

Reply:
I tested it:
```
/Users/gziolo/PhpstormProjects/gutenberg/packages/components/storybook/addons.js
  4:1  error  '@storybook/addon-a11y' should be listed in the project's dependencies. Run 'npm i -S @storybook/addon-a11y' to add it          import/no-extraneous-dependencies
  5:1  error  '@storybook/addon-viewport' should be listed in the project's dependencies. Run 'npm i -S @storybook/addon-viewport' to add it  import/no-extraneous-dependencies

/Users/gziolo/PhpstormProjects/gutenberg/packages/components/storybook/config.js
  4:1  error  '@storybook/react' should be listed in the project's dependencies. Run 'npm i -S @storybook/react' to add it            import/no-extraneous-dependencies
  5:1  error  '@storybook/addon-a11y' should be listed in the project's dependencies. Run 'npm i -S @storybook/addon-a11y' to add it  import/no-extraneous-dependencies
```

We can relax those checks for stories but not the storybook itself.